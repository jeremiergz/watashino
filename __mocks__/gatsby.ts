import Gatsby from 'gatsby';

module.exports = {
  ...Gatsby,
  graphql: jest.fn(),
  navigate: jest.fn(),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
};
