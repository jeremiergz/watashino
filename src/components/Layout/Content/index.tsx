import React, { HTMLAttributes } from 'react';
import Box from '../../primitives/Box';
import Heading from '../../widgets/Heading';
import SEO, { SEOProps } from '../../widgets/SEO';

type ContentProps = Pick<HTMLAttributes<HTMLElement>, 'children'> &
    Omit<SEOProps, 'title'> & {
        title?: string;
        type: 'post' | 'section';
    };

const typesMapping = {
    post: 'article',
    section: 'section',
} as const;

const Content = ({ children, description, keywords, lang, meta, title, type }: ContentProps) => {
    return (
        <>
            <SEO description={description} keywords={keywords} lang={lang} meta={meta} title={title} />
            {title && (
                <Heading marginTop={0} variant="h1">
                    {title}
                </Heading>
            )}
            <Box as={typesMapping[type]}>{children}</Box>
        </>
    );
};

Content.displayName = 'Content';

export default Content;
