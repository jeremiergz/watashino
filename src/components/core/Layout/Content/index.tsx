import React, { HTMLAttributes } from 'react';
import Heading from '../../../common/Heading';
import SEO, { SEOProps } from '../../../common/SEO';
import Box from '../../../primitives/Box';

type ContentProps = Pick<HTMLAttributes<HTMLElement>, 'children'> &
    Omit<SEOProps, 'title'> & {
        title?: string;
        type: 'post' | 'section';
    };

const typesMapping = {
    post: 'article',
    section: 'section',
} as const;

const Content = ({ children, description, keywords, lang, meta, title, type }: ContentProps) => {
    return (
        <>
            <SEO description={description} keywords={keywords} lang={lang} meta={meta} title={title} />
            {title && (
                <Heading marginTop={0} variant="h1">
                    {title}
                </Heading>
            )}
            <Box as={typesMapping[type]}>{children}</Box>
        </>
    );
};

Content.displayName = 'Content';

export default Content;
