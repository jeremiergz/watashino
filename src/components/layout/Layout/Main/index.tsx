import clsx from 'clsx';
import React from 'react';

function Main({ children, className, ...rest }: MainProps): JSX.Element {
  return (
    <main className={clsx(className, 'grid max-w-5xl transform-gpu will-change-transform')} {...rest}>
      {children}
    </main>
  );
}

export type MainProps = React.ComponentProps<'main'>;
export default Main;
