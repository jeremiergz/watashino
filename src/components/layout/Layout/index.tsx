import clsx from 'clsx';
import React from 'react';

function Layout({ children }: LayoutProps): JSX.Element {
  return (
    <div className={clsx('m-auto flex h-full w-full flex-col items-center pt-12 md:pt-0', 'dark:text-gray-300')}>
      {children}
    </div>
  );
}

export type LayoutProps = {
  children: React.ReactNode;
};
export default Layout;
