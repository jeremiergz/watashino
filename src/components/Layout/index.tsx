import Footer from 'components/Footer';
import Header from 'components/Header';
import Box from 'components/primitives/Box';
import { useTheming } from 'components/providers/ThemeProvider';
import useInstallPWA from 'hooks/useInstallPWA';
import React, { HTMLAttributes } from 'react';
import { GlobalStyle } from 'theme';
import Content from './Content';

const Layout: React.FC<LayoutProps> & { Content: typeof Content } = ({ children }) => {
  const { theme } = useTheming();
  useInstallPWA();
  return (
    <>
      <GlobalStyle />
      <Header />
      <Box
        as="main"
        margin="auto"
        maxWidth={theme.breakpoints[3]}
        paddingBottom={{ _: 78, tablet: 108 }}
        paddingX={{ _: 16, tablet: 32 }}
        textAlign="center"
      >
        {children}
      </Box>
      <Footer />
    </>
  );
};

Layout.Content = Content;
Layout.displayName = 'Layout';

export type LayoutProps = HTMLAttributes<HTMLDivElement>;
export default Layout;
