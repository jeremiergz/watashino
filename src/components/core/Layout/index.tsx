import React, { HTMLAttributes, useState } from 'react';
import { createGlobalStyle, ThemeProvider } from 'styled-components';
import { DarkTheme, LightTheme, Theme } from '../../../theme';
import Box from '../../primitives/Box';
import Footer from '../Footer';
import Header from '../Header';
import ThemeToggle from '../ThemeToggle';
import Content from './Content';

type LayoutProps = HTMLAttributes<HTMLDivElement>;

const GlobalStyle = createGlobalStyle`
    html, body {
        background-color: ${({ theme }: { theme: Theme }) => theme.background};
        height: 100%;
        width: 100%;
        margin: 0;
        padding: 0;
        > * {
            color: ${({ theme }: { theme: Theme }) => theme.colors.text};
            font-family: ${({ theme }: { theme: Theme }) => theme.fonts.main};
            font-weight: ${({ theme }: { theme: Theme }) => theme.fontWeights.regular};
        }
    }
    #___gatsby {
        position: relative;
        min-height: 100%;
    }
    /* Format Markdown images loaded by gatsby-remark-images */
    .gatsby-resp-image-wrapper {
        .gatsby-resp-image-background-image, .gatsby-resp-image-image {
            border-radius: 8px;
        }
    }
`;

const Layout = ({ children }: LayoutProps) => {
  const isBrowser = typeof window !== 'undefined';
  const storedTheme = (isBrowser && (localStorage.getItem('theme') as ThemeType)) || 'light';
  const [theme, setTheme] = useState<ThemeType>(storedTheme);
  const saveTheme = (type: ThemeType) => {
    setTheme(type);
    if (isBrowser) localStorage.setItem('theme', type);
  };
  const selectedTheme = theme === 'light' ? LightTheme : DarkTheme;
  return (
    <ThemeProvider theme={selectedTheme}>
      <>
        <ThemeToggle setTheme={saveTheme} />
        <Header />
        <Box
          as="main"
          margin="auto"
          maxWidth={selectedTheme.breakpoints[3]}
          paddingBottom={{ _: 78, tablet: 108 }}
          paddingX={{ _: 16, tablet: 32 }}
          textAlign="center"
        >
          {children}
        </Box>
        <Footer />
        <GlobalStyle />
      </>
    </ThemeProvider>
  );
};

Layout.displayName = 'Layout';

Layout.Content = Content;

export { LayoutProps };
export default Layout;
