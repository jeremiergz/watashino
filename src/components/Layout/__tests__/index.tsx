import * as Gatsby from 'gatsby';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Layout from '..';

describe('components/Layout Test Suite', () => {
    beforeAll(() => {
        const nodes = [
            {
                icon: 'icon',
                ignoreInNavigation: false,
                keywords: [],
                name: 'Test',
                navOrder: 0,
                to: '/',
            },
        ];
        const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
        useStaticQuery
            .mockImplementationOnce(() => ({
                allNavigationJson: { nodes },
                dataJson: { jobTitle: 'Full Stack Engineer' },
                file: {
                    childImageSharp: {
                        fixed: {
                            base64: 'data:image/png;base64',
                            aspectRatio: 1,
                            src: '/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png',
                            srcSet:
                                '/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png 1x,\n/static/89ee547c5db6cbf9b37a682e8b3f3f57/1e576/profile-pic.png 1.5x,\n/static/89ee547c5db6cbf9b37a682e8b3f3f57/6cb10/profile-pic.png 2x',
                        },
                    },
                },
            }))
            .mockImplementationOnce(() => ({
                allNavigationJson: { nodes },
                site: {
                    siteMetadata: {
                        title: 'SEO Test title',
                        description: 'SEO Test description.',
                        author: 'SEO Test author',
                    },
                },
            }));
    });

    it('matches snapshot', () => {
        const jsx = <Layout>Layout Component</Layout>;
        expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
    });
});
