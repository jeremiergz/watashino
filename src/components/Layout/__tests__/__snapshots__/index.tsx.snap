// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Layout Test Suite matches snapshot 1`] = `
Array [
  <header
    className="sc-bdVaJa kOkKhQ"
    display="flex"
  >
    <a
      className="sc-bdVaJa sc-bxivhb jKEtLG"
      href="/about-me"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "display": "inline-block",
            "height": undefined,
            "overflow": "hidden",
            "position": "relative",
            "width": undefined,
          }
        }
      >
        <img
          alt=""
          src="data:image/png;base64"
          style={
            Object {
              "height": "80px",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "transitionDelay": "500ms",
              "width": "80px",
            }
          }
        />
        <picture>
          <source
            srcSet="/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png 1x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/1e576/profile-pic.png 1.5x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/6cb10/profile-pic.png 2x"
          />
          <img
            alt=""
            loading="lazy"
            onLoad={[Function]}
            src="/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png"
            srcSet="/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png 1x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/1e576/profile-pic.png 1.5x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/6cb10/profile-pic.png 2x"
            style={
              Object {
                "height": "80px",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "80px",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png 1x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/1e576/profile-pic.png 1.5x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/6cb10/profile-pic.png 2x\\" /><img loading=\\"lazy\\" srcset=\\"/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png 1x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/1e576/profile-pic.png 1.5x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/6cb10/profile-pic.png 2x\\" src=\\"/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
      <div
        className="sc-bdVaJa ezLUsa"
        color="text"
        display="flex"
      >
        <h3
          className="sc-bwzfXH sc-htpNat klUrKO"
          color="primary"
          fontWeight="bold"
          letterSpacing={-1}
        >
          Jeremie Rodriguez
        </h3>
        <h4
          className="sc-bwzfXH sc-htpNat bvVtPB"
          fontWeight="semi-bold"
        >
          Full Stack Engineer
        </h4>
      </div>
    </a>
    <nav
      className="sc-bdVaJa cRCMnF"
      display="flex"
    >
      <div
        className="sc-bdVaJa fvcPEV"
        display="flex"
      >
        <a
          className="sc-bdVaJa sc-bxivhb jXVeSa"
          href="/"
        >
          <div
            className="sc-bdVaJa ipxrgW"
            height={32}
            width={32}
          />
          Test
        </a>
      </div>
    </nav>
  </header>,
  <main
    className="sc-bdVaJa bTnvRn"
  >
    Layout Component
  </main>,
]
`;
